mne/io/neuralynx/neuralynx.py

```diff
@@ -188,15 +325,41 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
             -1, 0
         ]  # express stop sample relative to segment onset
 
+        # array containing Segments
+        segments_arr = np.array(neo_block[0].segments, dtype=object)
+
+        # if gaps were detected correctly insert gap Segments in between valid Segments
+        if self._raw_extras[0]["seg_gap_dict"]:
```

Comment:
... maybe my idea above about having a single code path won't work here. But if it does that's great!